network/wsNetwork_test.go

```diff
@@ -3757,3 +3759,47 @@ func TestWebsocketNetworkTelemetryTCP(t *testing.T) {
 	t.Log("closed detailsA", string(pcdA))
 	t.Log("closed detailsB", string(pcdB))
 }
+
+func TestMaxHeaderSize(t *testing.T) {
```

Comment:
could you add a test for the following:
1. Set wsMaxHeaderBytes to some value and have a peer to mock peer that sends exact `wsMaxHeaderBytes` bytes of headers + some body (not sure if this survives/allowed for upgrade request)
2. Check all harders and values are available in `response.Header` after `conn, response, err := websocketDialer.DialContext`
3. Check the client receives body bytes as well (if this works)

Reply:
Ok yeah that was the test that I mentioned I have in websocket. And for that I do need a mocked peer but I can do it since I have a mocked peer in the previous commit that I deleted after realizing that I didn't need it for this approach.